### ISSUE TITLE
Dask.read_parquet fails - distributed.protocol.pickle - INFO - Failed to deserialize

### ISSUE THREAD

**Issue by lostmygithubaccount** [dask/dask#5752]
I've been struggling to use dask.dataframe.read_parquet from a distributed client. Replacing the code with equivalent pandas code works fine. 

The code is at [https://github.com/lostmygithubaccount/dasky/blob/master/example.ipynb](https://github.com/lostmygithubaccount/dasky/blob/master/example.ipynb)

In short:

* download data to a cluster at '/mnt/noaa' in format '/mnt/noaa/year=*/month=*/*.parquet'
* can glob the files and see they are there
* can read in the parquet files using pandas.read_parquet
* cannot read in the parquet files using dask.dataframe.read_parquet

When I try to switch to Dask, I get a distributed.protocol.pickle error. 

```python
import dask
import dask.dataframe as dd
import pandas as pd
import glob

files = dask.delayed(glob.glob)(f'{path}/**/*.parquet', recursive=True).compute()
df = dd.from_delayed([dask.delayed(pd.read_parquet)(file) for file in files]) # this is fine

df = dask.delayed(dd.read_parquet)(files, engine='fastparquet').compute() # this fails
```

Traceback:
distributed.protocol.pickle - INFO - Failed to deserialize b'\x80\x04\x95\x8b\x0b\x00\x00\x00\x00\x00\x00\x8c\x16tblib.pickling_support\x94\x8c\x12unpickle_exception\x94\x93\x94(\x8c\x08builtins\x94\x8c\x11IsADirectoryError\x94\x93\x94K\x15\x8c\x0eIs a directory\x94\x8c\x13/mnt/noaa/year=2008\x94\x87\x94Nh\x00\x8c\x12unpickle_traceback\x94\x93\x94\x8c\x05tblib\x94\x8c\x05Frame\x94\x93\x94)\x81\x94}\x94(\x8c\tf_globals\x94}\x94(\x8c\x08__name__\x94\x8c\x12distributed.worker\x94\x8c\x08__file__\x94\x8ch/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/distributed/worker.py\x94u\x8c\x06f_code\x94h\x0b\x8c\x04Code\x94\x93\x94)\x81\x94}\x94(\x8c\x0bco_filename\x94\x8ch/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/distributed/worker.py\x94\x8c\x07co_name\x94\x8c\x0eapply_function\x94ububM\xf2\x0ch\x0b\x8c\tTraceback\x94\x93\x94)\x81\x94}\x94(\x8c\x08tb_frame\x94h\r)\x81\x94}\x94(h\x10}\x94(h\x12h\x13h\x14h\x15uh\x16h\x18)\x81\x94}\x94(h\x1bh\x1ch\x1d\x8c\x0cexecute_task\x94ubub\x8c\ttb_lineno\x94M\x8b\x0c\x8c\x07tb_next\x94h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12\x8c\ndask.utils\x94h\x14\x8c`/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/dask/utils.py\x94uh\x16h\x18)\x81\x94}\x94(h\x1b\x8c`/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/dask/utils.py\x94h\x1d\x8c\x05apply\x94ububh*K\x1dh+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12\x8c\x1edask.dataframe.io.parquet.core\x94h\x14\x8ct/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/dask/dataframe/io/parquet/core.py\x94uh\x16h\x18)\x81\x94}\x94(h\x1b\x8ct/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/dask/dataframe/io/parquet/core.py\x94h\x1d\x8c\x0cread_parquet\x94ububh*K\xe0h+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12\x8c%dask.dataframe.io.parquet.fastparquet\x94h\x14\x8c{/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/dask/dataframe/io/parquet/fastparquet.py\x94uh\x16h\x18)\x81\x94}\x94(h\x1b\x8c{/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/dask/dataframe/io/parquet/fastparquet.py\x94h\x1d\x8c\rread_metadata\x94ububh*K\xb2h+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12hGh\x14hHuh\x16h\x18)\x81\x94}\x94(h\x1bhKh\x1d\x8c\x13_determine_pf_parts\x94ububh*Kbh+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12\x8c\x0ffastparquet.api\x94h\x14\x8ce/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/fastparquet/api.py\x94uh\x16h\x18)\x81\x94}\x94(h\x1bh[h\x1d\x8c\x08__init__\x94ububh*K[h+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12\x8c\x10fastparquet.util\x94h\x14\x8cf/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/fastparquet/util.py\x94uh\x16h\x18)\x81\x94}\x94(h\x1bheh\x1d\x8c\x12metadata_from_many\x94ububh*K|h+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12hdh\x14heuh\x16h\x18)\x81\x94}\x94(h\x1bheh\x1d\x8c\n<listcomp>\x94ububh*K|h+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12hZh\x14h[uh\x16h\x18)\x81\x94}\x94(h\x1bh[h\x1dh^ububh*Ksh+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12\x8c\x0bfsspec.spec\x94h\x14\x8ca/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/fsspec/spec.py\x94uh\x16h\x18)\x81\x94}\x94(h\x1b\x8ca/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/fsspec/spec.py\x94h\x1d\x8c\x04open\x94ububh*M\xd4\x02h+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12\x8c\x1cfsspec.implementations.local\x94h\x14\x8cr/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/fsspec/implementations/local.py\x94uh\x16h\x18)\x81\x94}\x94(h\x1b\x8cr/azureml-envs/azureml_2637bd91ce20445caf0288ade38364e0/lib/python3.6/site-packages/fsspec/implementations/local.py\x94h\x1d\x8c\x05_open\x94ububh*Kkh+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12h\x88h\x14h\x89uh\x16h\x18)\x81\x94}\x94(h\x1bh\x8ch\x1dh^ububh*K\xadh+h )\x81\x94}\x94(h#h\r)\x81\x94}\x94(h\x10}\x94(h\x12h\x88h\x14h\x89uh\x16h\x18)\x81\x94}\x94(h\x1bh\x8ch\x1dh\x8dububh*K\xb2ububububububububububububub\x87\x94R\x94t\x94R\x94.'
Traceback (most recent call last):
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/pickle.py", line 59, in loads
    return pickle.loads(x)
AttributeError: Can't get attribute 'unpickle_exception' on <module 'tblib.pickling_support' from '/anaconda/envs/azureml_py36/lib/python3.6/site-packages/tblib/pickling_support.py'>
distributed.protocol.core - CRITICAL - Failed to deserialize
Traceback (most recent call last):
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/core.py", line 124, in loads
    value = _deserialize(head, fs, deserializers=deserializers)
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 268, in deserialize
    return loads(header, frames)
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 62, in pickle_loads
    return pickle.loads(b"".join(frames))
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/pickle.py", line 59, in loads
    return pickle.loads(x)
AttributeError: Can't get attribute 'unpickle_exception' on <module 'tblib.pickling_support' from '/anaconda/envs/azureml_py36/lib/python3.6/site-packages/tblib/pickling_support.py'>
distributed.utils - ERROR - Can't get attribute 'unpickle_exception' on <module 'tblib.pickling_support' from '/anaconda/envs/azureml_py36/lib/python3.6/site-packages/tblib/pickling_support.py'>
Traceback (most recent call last):
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/utils.py", line 662, in log_errors
    yield
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/client.py", line 1139, in _handle_report
    msgs = await self.scheduler_comm.comm.read()
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/comm/tcp.py", line 208, in read
    frames, deserialize=self.deserialize, deserializers=deserializers
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/tornado/gen.py", line 209, in wrapper
    yielded = next(result)
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/comm/utils.py", line 66, in from_frames
    res = _from_frames()
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/comm/utils.py", line 52, in _from_frames
    frames, deserialize=deserialize, deserializers=deserializers
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/core.py", line 124, in loads
    value = _deserialize(head, fs, deserializers=deserializers)
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 268, in deserialize
    return loads(header, frames)
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/serialize.py", line 62, in pickle_loads
    return pickle.loads(b"".join(frames))
  File "/anaconda/envs/azureml_py36/lib/python3.6/site-packages/distributed/protocol/pickle.py", line 59, in loads
    return pickle.loads(x)
AttributeError: Can't get attribute 'unpickle_exception' on <module 'tblib.pickling_support' from '/anaconda/envs/azureml_py36/lib/python3.6/site-packages/tblib/pickling_support.py'>

**Comment by mrocklin** [comment]
Thanks for raising an issue @lostmygithubaccount .  My guess is that you have non-uniform versions of libraries across your client and workers. You might try running the following to check:

```python
client.get_versions(check=True)
```

Other than that I'm not sure.  I took a brief look at your example notebook but it was long so I decided to pass.  If you end up needing more help then we would appreciate a minimal failing example.  See https://blog.dask.org/2018/02/28/minimal-bug-reports .  

I'm going to close this for now.  If the above doesn't work and you have something more narrowly attributable to Dask let us know and we'll reopen.
